File: examples/modulith-clinical-tool-jpa/src/test/java/io/zenwave360/example/clinicaltool/modules/surveys/config/RepositoriesInMemoryConfig.java
package io.zenwave360.example.clinicaltool.modules.surveys.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Primary;

import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.*;
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory.*;


//@Configuration
public class RepositoriesInMemoryConfig {


    protected final SurveyRepository surveyRepository = new SurveyRepositoryInMemory();
    @Bean @Primary
    public <T extends SurveyRepository> T surveyRepository() {
        return (T) surveyRepository;
    }
    protected final QuestionRepository questionRepository = new QuestionRepositoryInMemory();
    @Bean @Primary
    public <T extends QuestionRepository> T questionRepository() {
        return (T) questionRepository;
    }
    protected final SurveyAnswersRepository surveyAnswersRepository = new SurveyAnswersRepositoryInMemory();
    @Bean @Primary
    public <T extends SurveyAnswersRepository> T surveyAnswersRepository() {
        return (T) surveyAnswersRepository;
    }

}

File: examples/modulith-klinical-tool-jpa/src/test/kotlin/io/zenwave360/example/clinicaltool/modules/surveys/config/RepositoriesInMemoryConfig.kt
package io.zenwave360.example.clinicaltool.modules.surveys.config

import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.*
import io.zenwave360.example.clinicaltool.modules.surveys.repository.jpa.inmemory.*
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Primary

// @Configuration
open class RepositoriesInMemoryConfig {

    protected val surveyRepository = SurveyRepositoryInMemory()

    @Bean
    @Primary
    fun surveyRepository(): SurveyRepositoryInMemory {
        return surveyRepository
    }

    protected val questionRepository = QuestionRepositoryInMemory()

    @Bean
    @Primary
    fun questionRepository(): QuestionRepositoryInMemory {
        return questionRepository
    }

    protected val surveyAnswersRepository = SurveyAnswersRepositoryInMemory()

    @Bean
    @Primary
    fun surveyAnswersRepository(): SurveyAnswersRepositoryInMemory {
        return surveyAnswersRepository
    }
}
